### ISSUE TITLE
SC2086 false positive on builtin command exit.

### ISSUE THREAD

**Issue by nnathan** [koalaman/shellcheck#1730]
SC2086 (quoting to avoid word splitting) should be disabled when a parameter is supplied to the `exit` command since it is a built-in and only requires an integer parameter. It kind of looks yuck to quote it.

#### For bugs
- SC2086 
- 0.7.0 and online 
- [X ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Slight related to #289.

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
bad() {
  ret=$1
  shift
  echo -e failed: "$@"
  exit $ret
}
```

#### Here's what shellcheck currently says:

On the `exit $ret` line:
SC2086: Double quote to prevent globbing and word splitting.

#### Here's what I wanted or expected to see:

I'm thinking shellcheck should detect the exit command is being used and not apply the SC2086 rule.

**Comment by matthewpersico** [comment]
hmmm. Assuming you don’t do something inane like `bad ‘stupid text’`, we are talking about what happens when you call `bad “42 anything else”`. Your way will `exit 42 anything else` which might work. Shellcheck wants `exit ‘42 something else’`, which is completely bonkers.

**Comment by nnathan** [comment]
Exactly, you can't lint against an invalid parameter to exit in this case. Well you might be able to - but that would be incredibly tricky.

So assuming a valid input to exit, it then must be an integer. An integer doesn't need quotes with standard IFS. Ergo no need for SC2086.

**Comment by a1346054** [comment]
See #2341 for why this isn't a false positive.

This issue can be closed.

**Comment by nnathan** [comment]
> See #2341 for why this isn't a false positive.
> 
> This issue can be closed.

Disagree - and maybe I should be pointing this out on your issue instead. A linter should be "best-effort" and should try to capture the typical setup of a script. It shouldn't concern itself with extreme edge cases such as setting `IFS=0`.

**Comment by kurahaupo** [comment]
As I mentioned in #289, I'm inclined to think that this would be better resolved by having an option to ignore lack of quoting on numeric expansions (such as variables with the `-i` attribute), and then (when that's in effect) suggesting that the variable be declared with the `-i` attribute when used with commands that take integer parameters such as `shift`, `let`, `return`, and of course `exit`.
